src/Apps/CollectorProfile/Routes/Saves2/Utils/getSelectedCollectionIds.ts

```diff
@@ -0,0 +1,26 @@
+import { difference, uniq } from "lodash"
+
+interface CollectionNode {
+  isSavedArtwork: boolean
+  internalID: string
+}
+
+interface Options<T> {
+  collections: T[]
+  addToCollectionIDs: string[]
+  removeFromCollectionIDs: string[]
+}
+
+export const getSelectedCollectionIds = <T extends CollectionNode>(
```

Comment:
I'm not sure I follow the question — the check on L18 only works if you pass in the nodes, right?

Reply:
yes. I mean maybe take [this code](https://github.com/artsy/force/blob/f61bcd9efe288739e9cd0d29cddd284335a06e3a/src/Apps/CollectorProfile/Routes/Saves2/Utils/getSelectedCollectionIds.ts#L18-L19) outside of `getSelectedCollectionIds` and [here](https://github.com/artsy/force/blob/f61bcd9efe288739e9cd0d29cddd284335a06e3a/src/Apps/CollectorProfile/Routes/Saves2/Utils/getSelectedCollectionIds.ts#L9) instead of `CollectionNode[]` pass `string[]`